core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/CompactExpressionEvaluator.scala

```diff
@@ -429,55 +468,67 @@ class CompactExpressionEvaluator(repository: CompactEntityRepository) extends Ex
         }
       }
 
-      // For each expression in this query plan, create an ExpressionAndResult
-      plan.expressionMappings.toSeq.flatMap { case (expression, attributeNames) =>
-        attributeNames.map { attrName =>
-          val attributeName = AttributeName.fromDelimitedName(attrName)
-
-          val entityToAttributeValues: Map[String, Try[Seq[AttributeValue]]] = if (entityType == rootEntityType) {
-            // Group all results under the original entityName since we want results grouped by the starting entity type
-            val allAttrs: Seq[AttributeValue] = entityRecords.values.flatten.toSeq.flatMap { record =>
-              if (
-                attributeName == AttributeName.withDefaultNS(record.entityType + Attributable.entityIdAttributeSuffix)
-              ) {
-                Seq(AttributeString(record.name))
-              } else {
-                record.toEntity.attributes.get(attributeName) match {
-                  case Some(avl: AttributeValueList) =>
-                    avl.list
-                  case Some(av: AttributeValue) =>
-                    Seq(av)
-                  case _ =>
-                    Seq.empty
+      if (plan.expressionMappings.isEmpty) {
+        // Return entities with empty input resolutions
```

Comment:
This is the other main change - previously, since we were mapping over the expressions, if there weren't any, we just called the database, got the entities, and essentially threw them away.  This returns them anyway if there are no input expressions.